### DISCUSSION TITLE
Does Bochs aim for bit-exact x87 FPU operations?

### DISCUSSION THREAD

**Discussion by jgouly** [bochs-emu/Bochs#422] [Q&A]
I couldn't find an irc channel, so asking a question here.

I see Bochs has a softfloat implementation of fsin/fcos, do these match exactly the values that hardware would give?

**Comment by stlintel** [comment]
>I see Bochs has a softfloat implementation of fsin/fcos, do these match exactly the values that hardware would give?

They don't match the HW results exactly but instead Bochs always produce more precise results that real x86 HW would.
This is caused by fact that Bochs uses float128 with 112-bit mantissa for internal computations while real hardware use only 67-bit mantissa it computations.
